docs/conf.py

```diff
@@ -8,12 +8,36 @@
 from sys import path
 
 from setuptools_scm import get_version
+from setuptools_scm.git import fetch_on_shallow
+from setuptools_scm.git import parse
 
 
+# -- Special accommodations for RTD ------------------------------------------
+
+
+def parse_with_fetch(*args, **kwargs) -> str:
+    """If the repo is found to be shallow, fetch a full.
+
+    By default, RTD does a fetch --limit 50, if a tag is not
+    present in the last 50 commits, the version reported by setuptools_scm
+    will be incorrect and appears as ``v0.1.dev...`` in the towncrier changelog.
+    Another approach is to enable ``DONT_SHALLOW_CLONE`` for the repo
+    https://docs.readthedocs.io/en/stable/feature-flags.html#feature-flags
+    This was done for ansible-navigator on the day of this commit.
+
+    :param args: The arguments
+    :param kwargs: The keyword arguments
+    :returns: The parsed version
+    """
+    return parse(*args, pre_parse=fetch_on_shallow, **kwargs)
+
+
+get_scm_version = partial(get_version, parse=parse_with_fetch)
```

Comment:
Do we want to have this conditional for just RTD?

Reply:
I was thinking what if someone somehow managed to do this locally....what would the harm be, also, if the experimental func ever changes or acts up, it should be able to repro locally w/o having to set the magic RTD envvar

Reply:
Agreed. Just making sure all the questions are asked and answered :)